src/niri.rs

```diff
@@ -5137,6 +5265,60 @@ impl ClientData for ClientState {
     fn disconnected(&self, _client_id: ClientId, _reason: DisconnectReason) {}
 }
 
+impl WindowMRU {
+    fn new(niri: &mut Niri) -> Self {
+        // update the timestamp on the currently active window and
+        // prepare a new WindowMRU
+        niri.lockin_focus();
+
+        // and build a list of MappedId sorted by timestamp
+        let mut bh = BinaryHeap::new();
+
+        niri.layout.with_windows(|m, _, _| {
+            if let Some(t) = m.get_focus_timestamp() {
+                bh.push((t, m.id()))
+            }
+        });
+
+        WindowMRU {
+            ids: bh
+                .into_sorted_vec()
+                .into_iter()
+                .map(|(_, id)| id)
+                .rev()
+                .collect(),
+            current: 0,
+        }
+    }
+
+    fn mru_with<F>(&mut self, niri: &Niri, f: F) -> Option<Window>
+    where
+        // function that yields the index of the next element, arguments
+        // are (current_index, number of elements in the list)
+        F: Fn(usize, usize) -> usize,
```

Comment:
Kinda overcomplicated to do this with a function. Even just some kind of `reverse_order: bool` would work here